### PR TITLE
Update golden file names

### PR DESCRIPTION
**Pull request by zliu41**
Instead of `.plc.golden` for everything, now the golden files are named appropriately, e.g., `.pir.golden`, `.uplc.golden`, or just `.golden`.

### CODE REVIEW

**Review by effectfully — APPROVED** [review]
This is great.

File: plutus-core/plutus-core/test/TypeSynthesis/Golden/VerifySECP256k1Signature.plc.golden
Comment (michaelpj) [inline plutus-core/plutus-core/test/TypeSynthesis/Golden/VerifySECP256k1Signature.plc.golden:1]: should these files have disappeared entirely?

File: plutus-core/testlib/Test/Tasty/Extras.hs [hunk 1]
```diff
@@ -58,20 +58,20 @@ goldenVsDocM :: TestName -> FilePath -> IO (Doc ann) -> TestTree
...
+nestedGoldenVsText :: TestName -> FilePath -> Text -> TestNested
+nestedGoldenVsText name ext = nestedGoldenVsTextM name ext . pure
 
 -- | Check the contents of a file under a name prefix against a 'Text'.
-nestedGoldenVsTextM :: TestName -> IO Text -> TestNested
-nestedGoldenVsTextM name text = do
+nestedGoldenVsTextM :: TestName -> FilePath -> IO Text -> TestNested
+nestedGoldenVsTextM name ext text = do
     path <- ask
     -- TODO: make more generic
```
Comment (michaelpj) [inline plutus-core/testlib/Test/Tasty/Extras.hs:68]: you just did that

File: plutus-core/untyped-plutus-core/test/Evaluation/Golden/prod1.plc.golden [hunk 2]
```diff
@@ -1 +0,0 @@
-(Right (prod (con integer 11) (con integer 12)))
```
Comment (michaelpj) [inline plutus-core/untyped-plutus-core/test/Evaluation/Golden/prod1.plc.golden:1]: oh wow, this was my bad, these really were leftover golden files from a very old version of SOPs!

**Review by michaelpj — APPROVED** [review]
I'm guessing that all the deleted golden files really were dead, which is great.

File: plutus-core/untyped-plutus-core/test/DeBruijn/Golden/failBody0.plc.golden [hunk 3]
```diff
@@ -1 +0,0 @@
-(Left Free index: 999999)
```
[the remark this answers is not in the document]
Comment (zliu41) [reply plutus-core/untyped-plutus-core/test/DeBruijn/Golden/failBody0.plc.golden:1]: This test was moved to the `Strict` subdir.

File: plutus-core/plutus-core/test/TypeSynthesis/Golden/VerifySECP256k1Signature.plc.golden
Comment (zliu41) [reply plutus-core/plutus-core/test/TypeSynthesis/Golden/VerifySECP256k1Signature.plc.golden:1]: This test no longer exists.

**Pull request merged**